### PR TITLE
Backport dedupe fix and factor out stable_unique

### PR DESCRIPTION
**Pull request by jackgerrits**
Backport the fix for deduping command line arguments to fix performance degradation overtime from a large command line.

Application of following fix: https://github.com/JohnLangford/vowpal_wabbit/pull/1502

### CODE REVIEW

File: vowpalwabbit/global_data.cc [hunk 1]
```diff
@@ -201,6 +202,31 @@ std::vector<std::string> opts_to_args(const std::vector<boost::program_options::
 	return args;
 }
 
+// Performs unique operation over collection without requiring the collection to be sorted first.
+// Returns pointer to first element after unique elements to erase from until end of collection.
+// Original ordering of elements is preserved.
+// This function was extracted based on this implementation in: https://github.com/JohnLangford/vowpal_wabbit/pull/1502
+template <typename ForwardIterator>
```
Comment (rajan-chari) [inline vowpalwabbit/global_data.cc]: A couple of unit tests would be nice but we don't have good spot in the repo.

Comment (jackgerrits) [reply vowpalwabbit/global_data.cc]: Definitely agree! Will do this as part of merging this into John's VW

Comment (lokitoth) [reply vowpalwabbit/global_data.cc]: I think the decision here is to hotfix this and bring it over to JohnLangford/vowpal_wabbit:master along with unit testing going forward.

**Comment by lokitoth** [comment]
Looks good. Let's make sure we push this factoring out of the logic to JohnLangford:master, too, though it would probably be a separate change, for simplicity.

**Pull request merged**